packages/fetch-and-compile/lib/recognizer.ts

```diff
@@ -39,9 +39,11 @@ export class SingleRecognizer implements Recognizer {
     return this.recognized ? undefined : this.address;
   }
 
-  markUnrecognizable(address: string, reason?: FailureType): never {
+  markUnrecognizable(address: string, reason?: FailureType, error?: Error): never {
     //just throw...
-    if (reason) {
+    if (error) {
+      throw error;
+    } else if (reason) {
```

Comment:
This seems ok if `reason` can occur without `error`, but I may be misunderstanding the logic.

Reply:
There is one way at this can occur at the moment: If you try to fetch-and-compile a Vyper source, which this package doesn't support at the moment, then you'll end up here with `reason` equal to `"language"` but `error` not set (since nothing threw an actual error to rethrow).

Reply:
Thanks for clarifying that.

Reply:
would this be a good time to use an AggregateError? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AggregateError (polyfill for node < 15: https://www.npmjs.com/package/aggregate-error)

Reply:
Ganache uses one if multiple things go wrong on start up: https://github.com/trufflesuite/ganache/blob/956df4af117e3baa255abba4f4a38dee78eb6e64/src/packages/core/src/server.ts#L256

Reply:
Fascinating!  I didn't know about that.  Yes, I think it might!  But do you mind if I save that for a later PR?  I'm not entirely sure offhand how to integrate that into this well.

Reply:
Oh, hm, looks like `AggregateError` is only available on Node 15 and later.  Seems to me like that kind of rules it out for now... (how did you get away with using it in Ganache?)

Reply:
Oh duh the polyfill you linked.  Silly me.

Reply:
That is really nice @davidmurdoch

Reply:
Harry, yeah, I don't think it's super important, it's just kinda nice.